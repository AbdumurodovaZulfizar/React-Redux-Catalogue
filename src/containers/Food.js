import React, { useState } from "react";
import { Link } from "react-router-dom";
import PropTypes from 'prop-types';

const FoodItem = ({ food }) => {
  const [expanded, setExpanded] = useState(false);
  
  const item = { ...food };
  const handleExpandClick = () => {
    setExpanded(!expanded);
  };
  return (
    <div className="col-12 col-md-4 p-0">
      <Link
      to={{
        pathname: '/DetailPage/{item.idMeal}',
        state: { food: item },
      }}

      className="m-2 nav-link p-0 my_card"
    >
      <div onClick={handleExpandClick} className="m-2">
        <img alt="food" src={item.strMealThumb} className="landing-page-img"/>
        <div>
          <h5 className="text-center card_link">{item.strMeal}</h5>
        </div>
      </div>
    </Link>
    </div>
  );
};

export default FoodItem;

FoodItem.defaultProps = {
  food: {},
};

FoodItem.propTypes = {
  food: PropTypes.shape({}),
};

