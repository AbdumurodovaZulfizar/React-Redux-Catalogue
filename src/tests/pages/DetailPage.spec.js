import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Food from '../../containers/Food'

test('should render filterpage page correctly', () => {
  const component = renderer.create(
  <BrowserRouter>
  <Food />
  </BrowserRouter>
    );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders correctly only one food', () => {
  const food = {
    strName: 'FoodName',
    strId: '111',
  };

  const component = renderer.create(
    <BrowserRouter>
    <Food item={food} />
    </BrowserRouter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
