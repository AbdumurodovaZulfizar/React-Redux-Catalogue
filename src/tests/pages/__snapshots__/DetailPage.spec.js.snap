// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly only one food 1`] = `
<div
  className="col-12 col-md-4 p-0"
>
  <a
    className="m-2 nav-link p-0 my_card"
    href="/DetailPage/{item.idMeal}"
    onClick={[Function]}
  >
    <div
      className="m-2"
      onClick={[Function]}
    >
      <img
        alt="food"
        className="landing-page-img"
      />
      <div>
        <h5
          className="text-center card_link"
        />
      </div>
    </div>
  </a>
</div>
`;

exports[`should render filterpage page correctly 1`] = `
<div
  className="col-12 col-md-4 p-0"
>
  <a
    className="m-2 nav-link p-0 my_card"
    href="/DetailPage/{item.idMeal}"
    onClick={[Function]}
  >
    <div
      className="m-2"
      onClick={[Function]}
    >
      <img
        alt="food"
        className="landing-page-img"
      />
      <div>
        <h5
          className="text-center card_link"
        />
      </div>
    </div>
  </a>
</div>
`;
